CHANGELOG.md

```diff
@@ -32,9 +32,10 @@ And please only add new entries to the top of this list, right below the `# Unre
 - On Wayland, fix polling during consecutive `EventLoop::run_return` invocations.
 - On Windows, fix race issue creating fullscreen windows with `WindowBuilder::with_fullscreen`
 - On Android, `virtual_keycode` for `KeyboardInput` events is now filled in where a suitable match is found.
-- **Breaking:** Added new `WindowEvent::IME` supported on macOS, X11, and Wayland.
-- Added `Window::set_ime_allowed` supported on macOS, X11, and Wayland.
-- **Breaking:** IME input on macOS, X11, and Wayland won't be received unless it's explicitly allowed via `Window::set_ime_allowed` and new `WindowEvent::IME` events are handled.
+- **Breaking:** Added new `WindowEvent::IME` supported on macOS, Windows, X11, and Wayland.
+- Added `Window::set_ime_allowed` supported on macOS, Windows, X11, and Wayland.
+- **Breaking:** IME input on macOS, Windows, X11, and Wayland won't be received unless it's explicitly allowed via `Window::set_ime_allowed` and new `WindowEvent::IME` events are handled.
+- **Breaking:** On Windows, IME-drawn composing text won't be rendered and the behavior is the same with the other platforms.
```

Comment:
Are you talking about preedit text here? I don't quite understand that statement to be fair. Could you go a bit into details wrt that change and why it's breaking and what IME-drawn composing text?

Reply:
<details>
<summary></summary>

```diff
diff --git a/src/platform_impl/windows/event_loop.rs b/src/platform_impl/windows/event_loop.rs
index 290d6847..2c15aa16 100644
--- a/src/platform_impl/windows/event_loop.rs
+++ b/src/platform_impl/windows/event_loop.rs
@@ -34,7 +34,7 @@ use windows_sys::Win32::{
     UI::{
         Controls::{HOVER_DEFAULT, WM_MOUSELEAVE},
         Input::{
-            Ime::{GCS_COMPSTR, GCS_RESULTSTR, ISC_SHOWUICOMPOSITIONWINDOW},
+            Ime::{GCS_COMPSTR, GCS_RESULTSTR},
             KeyboardAndMouse::{
                 MapVirtualKeyA, ReleaseCapture, SetCapture, TrackMouseEvent, TME_LEAVE,
                 TRACKMOUSEEVENT, VK_F4,
@@ -1155,8 +1155,7 @@ unsafe fn public_window_callback_inner<T: 'static>(
                 }
             }
 
-            // Not calling DefWindowProc to hide composing text drawn by IME.
-            0
+            DefWindowProcW(window, msg, wparam, lparam)
         }
 
         WM_IME_ENDCOMPOSITION => {
@@ -1188,12 +1187,7 @@ unsafe fn public_window_callback_inner<T: 'static>(
             DefWindowProcW(window, msg, wparam, lparam)
         }
 
-        WM_IME_SETCONTEXT => {
-            // Hide composing text drawn by IME.
-            let wparam = wparam & (!ISC_SHOWUICOMPOSITIONWINDOW as usize);
-
-            DefWindowProcW(window, msg, wparam, lparam)
-        }
+        WM_IME_SETCONTEXT => DefWindowProcW(window, msg, wparam, lparam),
 
         // this is necessary for us to maintain minimize/restore state
         WM_SYSCOMMAND => {
diff --git a/src/platform_impl/windows/ime.rs b/src/platform_impl/windows/ime.rs
index ad797625..721732a0 100644
--- a/src/platform_impl/windows/ime.rs
+++ b/src/platform_impl/windows/ime.rs
@@ -11,9 +11,9 @@ use windows_sys::Win32::{
     UI::{
         Input::Ime::{
             ImmAssociateContextEx, ImmGetCompositionStringW, ImmGetContext, ImmReleaseContext,
-            ImmSetCandidateWindow, ATTR_TARGET_CONVERTED, ATTR_TARGET_NOTCONVERTED, CANDIDATEFORM,
-            CFS_EXCLUDE, GCS_COMPATTR, GCS_COMPSTR, GCS_CURSORPOS, GCS_RESULTSTR, IACE_CHILDREN,
-            IACE_DEFAULT,
+            ImmSetCompositionWindow, ATTR_TARGET_CONVERTED, ATTR_TARGET_NOTCONVERTED, CFS_POINT,
+            COMPOSITIONFORM, GCS_COMPATTR, GCS_COMPSTR, GCS_CURSORPOS, GCS_RESULTSTR,
+            IACE_CHILDREN, IACE_DEFAULT,
         },
         WindowsAndMessaging::{GetSystemMetrics, SM_IMMENABLED},
     },
@@ -120,14 +120,13 @@ impl ImeContext {
         }
         let (x, y) = spot.to_physical::<i32>(scale_factor).into();
 
-        let candidate_form = CANDIDATEFORM {
-            dwIndex: 0,
-            dwStyle: CFS_EXCLUDE,
+        let candidate_form = COMPOSITIONFORM {
+            dwStyle: CFS_POINT,
             ptCurrentPos: POINT { x, y },
             rcArea: zeroed(),
         };
 
-        ImmSetCandidateWindow(self.himc, &candidate_form);
+        ImmSetCompositionWindow(self.himc, &candidate_form);
     }
 
     pub unsafe fn set_ime_allowed(hwnd: HWND, allowed: bool) {

```

Reply:
![alacritty v0.10.1 and IME](https://user-images.githubusercontent.com/16546008/162600559-4b441b53-d128-4518-8442-b44fb6f96830.png)
This is alacritty v0.10.1 with IME. The text surrounded red rect in this is what I said "IME-drawn composing text".
On Windows, to enable to use IME in applications that use only WM_CHAR, the text is drawn automatically. Until this PR, winit had provided set_ime_position, but relied on this behavior when receiving text from IME.

![alacritty # 5790 with IME and this PR but showing IME-drawn composing text](https://user-images.githubusercontent.com/16546008/162600565-504a3ccf-b6b0-4361-b80d-e71f7bc7c6e9.png)
This is alacritty that uses IME events (alacritty/alacritty#5790), and winit that is removed code hiding IME-drawn composing text from this PR. Like this, same composing text draws twice, by application and Windows.

![alacritty # 5790 with IME and this PR](https://user-images.githubusercontent.com/16546008/162600594-4c33cd6b-9b3a-407d-a61f-2ef8652d50ab.png)
For applications using IME events, removing this will enable to provide consistency (=apps can draw and manage it) text input, but if applications support only Windows, relies on ReceivedCharacter and IME-drawn composing text, developers will have to support IME events newly like the other desktop platforms.
That's why I wrote this as breaking change.

Reply:
Oh, nice. Given that IME is disabled by default with the next release of winit, not sure that we should mention that in changelog at all?

The thing is that there won't be IME at all by default with the next release, and you should opt-in and handle events for it to work, so probably shouldn't mention that?

Reply:
Sorry, I missed this. I agree that I don't have to mention to it. I remove this.

Reply:
Sorry, I missed this. I agree that I don't have to mention this behavior. I'll remove last line.

Reply:
I removed it.